flexmeasures/utils/time_utils.py

```diff
@@ -85,7 +85,7 @@ def naturalized_datetime_str(dt: Optional[datetime]) -> str:
     local_dt = (
         dt.replace(tzinfo=pytz.utc).astimezone(local_timezone).replace(tzinfo=None)
     )
-    if dt >= datetime.utcnow() - timedelta(hours=24):
+    if naive_utc_from(dt) >= datetime.utcnow() - timedelta(hours=24):
```

Comment:
Three requests:

1. Good moment to expand the docstring. For example, in case dt is naive, do we assume it is utc or in the server timezone? And what is that 24 hours there for?

2. I'd also prefer to do the comparison with two aware datetimes (converted to one or the other, if needed) rather than with two naive datetimes, to reduce the risk of comparing two datetimes wrongly assumed to share a timezone.

3. Let's unittest this function with some cases. E.g. server time in utc or Amsterdam, dt in naive, utc or Seoul (and dt=None).

Reply:
I agree with points 1 and 3 and with the notion to spend some (limited) time now to make this util function better, but for point 2, we're comparing two UTC times ― not sure which edge cases you are thinking of.

As for 1: yes indeed, we are assuming UTC time if we have nothing to go on (see also `naive_utc_from`), as our database keeps UTC time. The other assumption would be to assume local time, which is an assumption that libraries do more often for naive datetimes.

Reply:
Great, thanks. I made 1 small commit myself here, too.